docs/standards/hcs-2.md

```diff
@@ -1,31 +1,206 @@
+
+
+### Memo for Indexers and Browsers
+
+A memo system is defined for indexers and browsers to understand the data's state and interpret it accordingly. The memo format follows:
+
+`[protocol_standard]:[indexed]`
+
+| Field     | Description                                                        | Example Value          |
+|-----------|--------------------------------------------------------------------|------------------------|
+| `protocol_standard`       | Protocol used by the registry, `hcs-2` for this standard.| `hcs-2`                |
+| `indexed`     | enum value of if all messages need pulled down or only the last / newest message | `0`           |
+
+
+| Indexed enum     | Description                                                       
+|-----------|-------------------------------------------------------------------------------------------|
+| `0`       | The topic id is indexed, and all messages should be read
+| `1`     | The topic id is not indexed, and only the last message should be used to determine state / topic data
+
+
+### Example Memo Format
+
+`hcs-2:0`
+
+### Indexed Registry Mechanics [enum: 0]
+Uses:
+Topic ids used for registiers where you need all records to get the data you need to execute logic. Typically good for most registeries. 
+
+Examples:
+1. User registry for profiles of a video game
+2. A registry for posts on a social media site for a specific user
+
+- Expectation for new records to be continually added.
+- Indexers should gather all files and metadata listed in the registry.
+- Processing state should start from the first message and proceed to the last sequential message number.
+
+
+### Non-Indexed Registry Mechanics [enum: 1]
+Uses:
+Topic ids used for dynamic state of an entity. The latest message being used is the current state of the entity being defined
+
+Examples:
+1. A description for a product on a ecom site
+2. A state of an NFT for a video game 
+
+- Expectation for new records to be continually added.
+- Indexers should gather `only the last message` and metadata in that message will determine the protocol and execution. 
+- Processing state should start from the first message and proceed to the last sequential message number.
+
+
+### Use Cases and Functionalities
+
+- Dynamic metadata enables NFTs and other assets to reflect changes and updates over time.
+- Full version history allows for a complete understanding of an item's evolution.
+- Enhanced file and website management through recursive indexing and state awareness.
+
+### Example URI Format
+
+Registry links should follow a consistent format to ensure easy access and interpretation:
+
+`hcs://2/{topic_id}`
+
+This facilitates direct access to specific registry entries and simplifies integration with external systems and applications.
+
+
+#### Register
+
+Register new entries or versions to the registry using the following JSON structure:
+
+```json
+{
```

Comment:
Could we introduce a `ttl` field to help support external infrastructure? I believe this will play well with DNS like registries in the future as well